core/src/main/java/com/datastrato/gravitino/storage/relational/service/UserMetaService.java

```diff
@@ -207,4 +208,24 @@ public <E extends Entity & HasIdentifier> UserEntity updateUser(
     }
     return newEntity;
   }
+
+  public int deleteUserMetasByLegacyTimeLine(long legacyTimeLine, int limit) {
+    AtomicInteger userDeletedCount = new AtomicInteger(0);
+    AtomicInteger userRoleRelDeletedCount = new AtomicInteger(0);
+
+    SessionUtils.doMultipleWithCommit(
+        () ->
+            userDeletedCount.set(
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    UserMetaMapper.class,
+                    mapper -> mapper.deleteUserMetasByLegacyTimeLine(legacyTimeLine, limit))),
+        () ->
+            userRoleRelDeletedCount.set(
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    UserRoleRelMapper.class,
+                    mapper ->
+                        mapper.deleteUserRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit))));
+
+    return userDeletedCount.get() + userRoleRelDeletedCount.get();
```

Comment:
What's the meaning of return value? Will we use the return value?

Reply:
We will delete in batches and use the return value to determine if there is no more data to delete.

https://github.com/datastrato/gravitino/blob/2bf1767ab71da9b1f55253de92c055899a82e515/core/src/main/java/com/datastrato/gravitino/storage/relational/RelationalGarbageCollector.java#L72

Reply:
If we just need to judge whether the count is zero or not, it's ok.